app/src/main/java/com/example/calculator_final/FragmentKeyboardFirst.kt

```diff
@@ -0,0 +1,59 @@
+package com.example.calculator_final
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import kotlinx.android.synthetic.main.fragment_keyboard_1.*
+
+class FragmentKeyboardFirst : Fragment() {
+
+    var communicator: Communicator? = null
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        communicator = context as Communicator
+        return inflater.inflate(R.layout.fragment_keyboard_1, container, false)
+    }
+
+    override fun onResume() {
+        super.onResume()
+
+        // Number
+        btnNumZero.setOnClickListener { communicator?.respondData("number", "0") }
```

Comment:
```suggestion
        btnNumZero.setOnClickListener { communicator?.respondData(ButtonType.NUMBER, "0") }
```
Tạo 1 enum để phân loại cho dễ nhé

Reply:
Vâng ạ